cmd/config-versions.go

```diff
@@ -722,3 +723,39 @@ type serverConfigV27 struct {
 	// Logger configuration
 	Logger loggerConfig `json:"logger"`
 }
+
+type authProviders struct {
+	RSA provider.RSAArgs `json:"rsa"`
+}
+
+// serverConfigV28 is just like version '27', stores additionally
+// STS auth providers.
+// IMPORTANT NOTE: When updating this struct make sure that
+// serverConfig.ConfigDiff() is updated as necessary.
+type serverConfigV28 struct {
+	quick.Config `json:"-"` // ignore interfaces
+
+	Version string `json:"version"`
+
+	// S3 API configuration.
+	Credential auth.Credentials `json:"credential"`
+	Region     string           `json:"region"`
+	Browser    BoolFlag         `json:"browser"`
+	Worm       BoolFlag         `json:"worm"`
+	Domain     string           `json:"domain"`
+
+	// Storage class configuration
+	StorageClass storageClassConfig `json:"storageclass"`
+
+	// Cache configuration
+	Cache CacheConfig `json:"cache"`
+
+	// Notification queue configuration.
+	Notify notifier `json:"notify"`
+
+	// Logger configuration
+	Logger loggerConfig `json:"logger"`
+
+	// Authentication providers.
+	AuthProviders authProviders `json:"auth"`
```

Comment:
can you check if curl -k https://localhost:9443/oauth2/jwks is a better way to get the Public key (i.e i dont know if other providers provide the same endpoint)

Reply:
nope, we need the RSA in a *rsa.PublicKey format, I can, of course, get it from this style 

```
openssl s_client -connect localhost:9443 -showcerts </dev/null 2>/dev/null | openssl x509 -outform PEM
```

And also use `go` package equivalents. Also for every JWT validation, we shouldn't or need to talk to an identity provider to get the certs. The current implementation is more generic here.